cost_estimation.go

```diff
@@ -0,0 +1,125 @@
+	client *Client
+}
+
+// CostEstimationStatus represents a costEstimation state.
+type CostEstimationStatus string
+
+//List all available costEstimation statuses.
+const (
+	CostEstimationCanceled CostEstimationStatus = "canceled"
+	CostEstimationErrored  CostEstimationStatus = "errored"
+	CostEstimationFinished CostEstimationStatus = "finished"
+	CostEstimationQueued   CostEstimationStatus = "queued"
+)
+
+// CostEstimation represents a Terraform Enterprise costEstimation.
+type CostEstimation struct {
+	ID               string                          `jsonapi:"primary,cost-estimations"`
+	LogReadURL       string                          `jsonapi:"attr,log-read-url"`
+	Status           CostEstimationStatus            `jsonapi:"attr,status"`
+	StatusTimestamps *CostEstimationStatusTimestamps `jsonapi:"attr,status-timestamps"`
+	ErrorMessage     string                          `jsonapi:"attr,error-message"`
+	// Costs to be implemented once it's finalized for GA
+	// Costs            *CostEstimationCosts            `jsonapi:"attr,costs"`
+}
+
+// CostEstimationStatusTimestamps holds the timestamps for individual costEstimation statuses.
+type CostEstimationStatusTimestamps struct {
+	CanceledAt time.Time `json:"canceled-at"`
+	ErroredAt  time.Time `json:"errored-at"`
+	FinishedAt time.Time `json:"finished-at"`
+	QueuedAt   time.Time `json:"queued-at"`
+}
+
+// Read a costEstimation by its ID.
+func (s *costEstimations) Read(ctx context.Context, costEstimationID string) (*CostEstimation, error) {
+	if !validStringID(&costEstimationID) {
+		return nil, errors.New("invalid value for costEstimation ID")
+	}
+
+	u := fmt.Sprintf("cost-estimations/%s", url.QueryEscape(costEstimationID))
+	req, err := s.client.newRequest("GET", u, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	p := &CostEstimation{}
+	err = s.client.do(ctx, req, p)
+	if err != nil {
+		return nil, err
+	}
+
+	return p, nil
+}
+
+// Logs retrieves the logs of a costEstimation.
+func (s *costEstimations) Logs(ctx context.Context, costEstimationID string) (io.Reader, error) {
+	if !validStringID(&costEstimationID) {
+		return nil, errors.New("invalid value for costEstimation ID")
+	}
+
+	// Get the costEstimation to make sure it exists.
+	p, err := s.Read(ctx, costEstimationID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Return an error if the log URL is empty.
+	if p.LogReadURL == "" {
+		return nil, fmt.Errorf("costEstimation %s does not have a log URL", costEstimationID)
+	}
+
+	u, err := url.Parse(p.LogReadURL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid log URL: %v", err)
+	}
+
+	done := func() (bool, error) {
+		p, err := s.Read(ctx, p.ID)
+		if err != nil {
+			return false, err
+		}
+
+		switch p.Status {
+		case CostEstimationCanceled, CostEstimationErrored, CostEstimationFinished:
+			return true, nil
+		default:
+			return false, nil
+		}
+	}
+
+	return &LogReader{
+		client: s.client,
+		ctx:    ctx,
+		done:   done,
+		logURL: u,
+	}, nil
+}
```

Comment:
I think the big question here is if this log will be a streaming log like the plan and apply logs are, of that it will be more like the policy-check logs. Looking at how things work, I think it is closer to the latter which means this will not give the desired result.